dev/com.ibm.ws.jdbc/src/com/ibm/ws/rsadapter/impl/WSManagedConnectionFactoryImpl.java

```diff
@@ -1426,8 +1427,9 @@ public final int getLoginTimeout() throws SQLException {
             if(!Driver.class.equals(type)) {
                 return ((CommonDataSource) dataSourceOrDriver).getLoginTimeout();
             }
-            //TODO behavior for java.sql.Driver
-            return 0;
+            //When using Driver, loginTimeout is blocked from being set via URL or properties
+            //Therefore we know the loginTimeout being used is from the DriverManager
+            return DriverManager.getLoginTimeout();
```

Comment:
Returning 0 here was actually correct. This means the system is defaulting the value, as is expected when we have not permitted the configuration of it to a non-zero value.  Even though DriverManager.getLoginTimeout value might end up being used as the default, note that the value can change over time and it would be very confusing to the user. If they see a value, for example of 60 here, it implies they configured it to 60 and that the value will remain 60.  The former is untrue and the latter cannot be guaranteed.